### PR TITLE
Feature/46

### PR DESCRIPTION
**Pull request by sglover**

### CODE REVIEW

File: ignition/service/framework.py [hunk 1]
```diff
@@ -193,7 +196,7 @@ def __register_service_capabilities(self, service_class):
 
     def __register_service_requirements(self, service_class, required_capabilities):
         for requirement_name, required_capability_class in required_capabilities.items():
-            if not issubclass(required_capability_class, Capability):
+            if required_capability_class is not None and not issubclass(required_capability_class, Capability):
```
Comment (dvaccarosenna) [inline ignition/service/framework.py]: When would the required_capability_class be None?

File: ignition/service/infrastructure.py [hunk 2]
```diff
@@ -267,11 +292,28 @@ def __init__(self, **kwargs):
...
+                'properties': properties,
+                'system_properties': system_properties,
+                'deployment_location': deployment_location
+            })
+        else:
+            create_response = self.driver.create_infrastructure(template, template_type, PropValueMap(system_properties), PropValueMap(properties), deployment_location)
+            if self.async_enabled is True:
+                self.__async_infrastructure_task_completion(create_response.infrastructure_id, create_response.request_id, deployment_location)
+
         return create_response
```
Comment (dvaccarosenna) [inline ignition/service/infrastructure.py]: if self.async_requests_enabled==True won't the create_response be None?

File: ignition/service/framework.py (hunk 1 above)
Comment (sglover) [reply ignition/service/framework.py]: Added in error, will remove.

File: ignition/service/infrastructure.py (hunk 2 above)
Comment (sglover) [reply ignition/service/infrastructure.py]: Yep, I'll fix this up. Good spot!

**Pull request merged**